WordPressKitTests/ReaderPostServiceRemote+CardsTests.swift

```diff
@@ -93,4 +93,47 @@ class ReaderPostServiceRemoteCardTests: RemoteTestCase, RESTTestable {
 
         XCTAssertTrue(mockRemoteApi.URLStringPassedIn?.contains("&page_handle=foobar") ?? false)
     }
+    
+    // Calls the API with the given sorting option
+    //
+    func testCallAPIWithTheGivenSortingOption() {
+        let readerPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: mockRemoteApi)
+
+        readerPostServiceRemote.fetchCards(for: [], sortingOption: "foobar", success: { _, _ in }, failure: { _ in })
+
+        XCTAssertTrue(mockRemoteApi.URLStringPassedIn?.contains("sort=foobar") ?? false)
+    }
+    
+    // Calls the API without the given sorting option
+    //
+    func testCallAPIWithoutTheGivenSortingOption() {
+        let readerPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: mockRemoteApi)
+
+        readerPostServiceRemote.fetchCards(for: [], sortingOption: nil, success: { _, _ in }, failure: { _ in })
+        
+        XCTAssertFalse(mockRemoteApi.URLStringPassedIn?.contains("sort=") ?? false)
+    }
+    
+    // Calls the API with "date" as a sorting option and checks if posts are ordered properly
+    //
+    func testCallAPIWithDateAsGivenSortOption() {
+        let expect = expectation(description: "Get cards sorted by date")
+        stubRemoteResponse("read/tags/cards?tags%5B%5D=cats&sort=date", filename: "reader-cards-success.json", contentType: .ApplicationJSON)
+        
+        readerPostServiceRemote.fetchCards(for: ["cats"], sortingOption: "date", success: { cards, _ in
+            let posts = cards.filter { $0.type == .post }
+            for i in 1..<posts.count {
+                guard let firstPostDate = posts[i-1].post?.sortDate,
+                      let secondPostDate = posts[i].post?.sortDate,
+                      firstPostDate > secondPostDate else {
+                    XCTFail("Posts should be sorted by date, starting with most recent post")
+                    expect.fulfill()
```

Comment:
should this expectation be fulfilled here?

Reply:
I see in the project that some test call the `fullfill()` after `XCTFail()` and some before. I thought that it is usual to fulfill the expectation at the end of checking the logic. What should be the correct way?

Reply:
I would agree. 👍
Looking at the other tests in the project we should remove the `fullfill()` call in a lot of tests after `XCTFail()`, maybe that can be a thing good to do in the future. 👌